From e51d4d3b88af00d6667f2055087ebfc47fb3107c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Sur=C3=BD?= <ondrej@isc.org>
Date: Tue, 7 Jan 2025 15:22:40 +0100
Subject: [PATCH] Isolate using the -T noaa flag only for part of the resolver
 test

Instead of running the whole resolver/ns4 server with -T noaa flag,
use it only for the part where it is actually needed.  The -T noaa
could interfere with other parts of the test because the answers don't
have the authoritative-answer bit set, and we could have false
positives (or false negatives) in the test because the authoritative
server doesn't follow the DNS protocol for all the tests in the resolver
system test.
---
 bin/tests/system/resolver/ns4/named.noaa | 12 ------------
 bin/tests/system/resolver/tests.sh       |  8 ++++++++
 2 files changed, 8 insertions(+), 12 deletions(-)
 delete mode 100644 bin/tests/system/resolver/ns4/named.noaa

This fixes CVE-2024-11187. Dependency Patch #1.

Signed-off-by: Javier Campos <jcampos@mvista.com>

Index: bind-9.10.5-P3/bin/tests/system/conf.sh.in
===================================================================
--- bind-9.10.5-P3.orig/bin/tests/system/conf.sh.in
+++ bind-9.10.5-P3/bin/tests/system/conf.sh.in
@@ -126,6 +126,18 @@ fi
 
 PYTHON=@PYTHON@
 
+start_server() {
+    $PERL "$SYSTEMTESTTOP/start.pl" "$SYSTESTDIR" "$@"
+}
+
+stop_server() {
+    $PERL "$SYSTEMTESTTOP/stop.pl" "$SYSTESTDIR" "$@"
+}
+
+send() {
+    $PERL "$SYSTEMTESTTOP/send.pl" "$@"
+}
+
 #
 # Determine if we support various optional features.
 #
Index: bind-9.10.5-P3/bin/tests/system/resolver/ns4/named.noaa
===================================================================
--- bind-9.10.5-P3.orig/bin/tests/system/resolver/ns4/named.noaa
+++ /dev/null
@@ -1,6 +0,0 @@
-Copyright (C) 2010  Internet Systems Consortium, Inc. ("ISC")
-See COPYRIGHT in the source root or http://isc.org/copyright.html for terms.
-
-$Id: named.noaa,v 1.2 2010/09/15 12:07:56 marka Exp $
-
-Add -T noaa.
Index: bind-9.10.5-P3/bin/tests/system/resolver/tests.sh
===================================================================
--- bind-9.10.5-P3.orig/bin/tests/system/resolver/tests.sh
+++ bind-9.10.5-P3/bin/tests/system/resolver/tests.sh
@@ -283,6 +283,10 @@ grep "status: NXDOMAIN" dig.ns5.out.${n}
 if [ $ret != 0 ]; then echo "I:failed"; fi
 status=`expr $status + $ret`
 
+stop_server ns4
+touch ns4/named.noaa
+start_server --noclean --restart --port ${PORT} ns4 || ret=1
+
 n=`expr $n + 1`
 echo "I:check that replacement of additional data by a negative cache no data entry clears the additional RRSIGs ($n)"
 ret=0
@@ -311,6 +315,10 @@ status=`expr $status + $ret`
 if [ $ret != 0 ]; then echo "I:failed"; ret=1; fi
 status=`expr $status + $ret`
 
+stop_server ns4
+rm ns4/named.noaa
+start_server --noclean --restart --port ${PORT} ns4 || ret=1
+
 n=`expr $n + 1`
 echo "I:checking that update a nameservers address has immediate effects ($n)"
 ret=0
